x/ccv/consumer/keeper/relay_test.go

```diff
@@ -263,22 +312,93 @@ func incrementTimeBy(s *KeeperTestSuite, jumpPeriod time.Duration) {
 	}
 }
 
-func (suite *KeeperTestSuite) TestOnAcknowledgement() {
+// TestOnAcknowledgementPacket tests application logic for acknowledgments of sent VSCMatured and Slash packets
+// in conjunction with the ibc module's execution of "acknowledgePacket",
+// according to https://github.com/cosmos/ibc/tree/main/spec/core/ics-004-channel-and-packet-semantics#processing-acknowledgements
+func TestOnAcknowledgementPacket(t *testing.T) {
+
+	// Channel ID to some dest chain that's not the established provider
+	channelIDToDestChain := "channelIDToDestChain"
+
+	// Channel ID to established provider
+	channelIDToProvider := "channelIDToProvider"
+
+	// Channel ID on destination (counter party) chain
+	channelIDOnDest := "ChannelIDOnDest"
+
+	// Instantiate custom keeper with mocks
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	cdc, storeKey, paramsSubspace, ctx := testkeeper.SetupInMemKeeper(t)
+
+	mockScopedKeeper := testkeeper.NewMockScopedKeeper(ctrl)
+	mockChannelKeeper := testkeeper.NewMockChannelKeeper(ctrl)
+
+	consumerKeeper := testkeeper.GetCustomConsumerKeeperWithMocks(
+		cdc,
+		storeKey,
+		paramsSubspace,
+		mockScopedKeeper,
+		mockChannelKeeper,
+		testkeeper.NewMockPortKeeper(ctrl),
+		testkeeper.NewMockConnectionKeeper(ctrl),
+		testkeeper.NewMockClientKeeper(ctrl),
+		testkeeper.NewMockSlashingKeeper(ctrl),
+		testkeeper.NewMockBankKeeper(ctrl),
+		testkeeper.NewMockAccountKeeper(ctrl),
+		testkeeper.NewMockIBCTransferKeeper(ctrl),
+		testkeeper.NewMockIBCCoreKeeper(ctrl),
+	)
+
+	// Set an established provider channel for later in test
+	consumerKeeper.SetProviderChannel(ctx, channelIDToProvider)
+
 	packetData := types.NewSlashPacketData(
 		abci.Validator{Address: bytes.HexBytes{}, Power: int64(1)}, uint64(1), stakingtypes.Downtime,
 	)
 
-	packet := channeltypes.NewPacket(packetData.GetBytes(), 1, providertypes.PortID, suite.path.EndpointB.ChannelID,
```

Comment:
This packet construction logic was changed. per [spec](https://github.com/cosmos/ibc/tree/main/spec/core/ics-004-channel-and-packet-semantics#processing-acknowledgements), it seems like ```acknowledgePacket``` executes on an ack of a packet originally sourced from this (consumer) chain. The provider chain's port/channel should not be "source" as existed here previously. Lmk if I'm misunderstanding the spec, or the original intention of this test